docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,413 @@
+Let's say some token has decimal of 12 and we want to swap 6 actual tokens. Here is how we can calculate input amount:
+6 \* 10^12 = 6000000000000.
+:::
+
+```typescript
+// --snip--
+
+const poolKey = newPoolKey(TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+
+// make sure that you swap in correct direction,
+// here we check if token0 (token we want to give) is tokenX
+const xToY = isTokenX(TOKEN0_ADDRESS, TOKEN1_ADDRESS)
+
+// here we want to swap 6 token0
+// token0 has 12 decimals so we need to multiply it by 10^12
+const amount = 6n * 10n ** 12n
+
+// approve token0 transfer
+await psp22.approve(account, invariant.contract.address.toString(), amount)
+
+// slippage is a price change you are willing to accept,
+// for examples if current price is 1 and your slippage is 1%, then price limit will be 1.01
+const allowedSlippage = toPercentage(1n, 3n) // 0.001 = 0.1%
+
+// swap message only accepts price limit so we have to calculate it in order to use slippage,
+// here we are passing current price and allowed slippage to calculate price limit
+const pool = await invariant.getPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+const sqrtPriceLimit = calculateSqrtPriceAfterSlippage(pool.sqrtPrice, allowedSlippage, !xToY)
+
+const result = await invariant.swap(account, poolKey, xToY, amount, true, sqrtPriceLimit)
+console.log(result.hash) // print transaction hash
+```
+
+### Query state
+
+```typescript
+// --snip--
+
+// query state
+const poolAfter = await invariant.getPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+const positionAfter = await invariant.getPosition(account, account.address, 0n)
+const lowerTickAfter = await invariant.getTick(account, poolKey, positionAfter.lowerTickIndex)
+const upperTickAfter = await invariant.getTick(account, poolKey, positionAfter.upperTickIndex)
+
+// calculate unclaimed fees
+const unclaimedFees = simulateUnclaimedFees(
+  poolAfter,
+  positionAfter,
+  lowerTickAfter,
+  upperTickAfter
+)
+
+// print amount of unclaimed x and y token
+console.log(unclaimedFees)
+```
+
+### Claim fees
+
+```typescript
+// --snip--
+
+// claim fees
+// specify position id, positions are indexed from 0
+const positionId = 0n
+await invariant.claimFee(account, positionId)
+
+// get balance of a specific token after claiming position fees and print it
+const accountBalance = await psp22.balanceOf(account, account.address)
+console.log(accountBalance)
+```
+
+### Remove position
+
+```typescript
+// remove position
+invariant.removePosition(account, positionId)
+
+// get balance of a specific token after removing position
+const accountToken0Balance = await psp22.balanceOf(account, account.address)
+await psp22.setContractAddress(TOKEN1_ADDRESS)
+const accountToken1Balance = await psp22.balanceOf(account, account.address)
+
+// print balances
+console.log(accountToken0Balance, accountToken1Balance)
+```
+
+### Wrap AZERO
```

Comment:
It's worth it to add psp22 section also. Especially explain token reloading with `setContractAddress` and fact that instance of PSP22 is not associated with any specific token, instead supports multiple tokens